source/features/useful-not-found-page.tsx

```diff
@@ -36,46 +38,156 @@ function parseCurrentURL(): string[] {
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
+async function findAndDisplayMissingPart(bar: Element): Promise<void> {
+	// The branch might have been deleted
+	const branchDeleted = await addBranchStatus(bar);
+	// Or the file exists, but on a different branch
+	const branchDoesNotHaveFile = await addDefaultBranchLink(bar);
+	// If either of the above are true, we don't need to go into the file's commit history.
+	if (branchDeleted || branchDoesNotHaveFile) {
+		return;
+	}
+
+	// The object might have been deleted/moved
+	if (await addObjectStatus(bar)) {
+		return;
+	}
+
+	// Or 410 Gone
+	await addCommitHistoryLink(bar);
+}
+
+async function addObjectStatus(bar: Element): Promise<boolean> {
+	// Get the file path from the parts
+	const parts = parseCurrentURL();
+	const filePath = parts.slice(4).join('/');
+
+	// Get the last 2 commits that include the file
+	const previousCommitsIncludingFile = await api.v3(`commits?path=${filePath}&per_page=2`);
+	// The latest commit will be the first object in the array
+	if (previousCommitsIncludingFile[0]) {
+		// Get a list of changes that happened in the repo with this commit
+		const lastCommitInfo = await api.v3(`commits/${previousCommitsIncludingFile[0].sha as string}`);
+
+		// Check what happened to this file
+		const fileInfo = lastCommitInfo.files.find((file: Record<string, string>) => {
+			if (file.filename === filePath) {
+				return true;
+			}
+
+			if (file.status === 'renamed' && file.previous_filename === filePath) {
+				return true;
+			}
+
+			return false;
+		});
+
+		const commitAuthor = lastCommitInfo.author.login;
+		const commitTime = new Date(lastCommitInfo.commit.committer.date);
+		const commitSha = lastCommitInfo.sha;
+		const shortenedCommitSha = commitSha.slice(0, 8);
+		const commitUrl = lastCommitInfo.html_url;
+
+		const urlToCommitAuthorProfile = lastCommitInfo.author.html_url;
+		const urlToLastBlob = fileInfo.blob_url;
+
+		// If it was removed, tell the user
+		if (fileInfo.status === 'removed') {
+			bar.after(
+				<p className="container mt-4 text-center">
+					The file you are looking for was deleted/moved by <a href={urlToCommitAuthorProfile}>{commitAuthor}</a> {twas(commitTime.getTime())} with commit <a href={commitUrl}>{shortenedCommitSha}</a>.
```

Comment:
We should use the `<relative-time>` element instead of twas. Refer to its usage from other parts of the extension. No imports are necessary.

Reply:
Will do, thanks.